File: 1993-sum-of-all-subset-xor-totals/sum-of-all-subset-xor-totals.py
class Solution:
    def subsetXORSum(self, nums: List[int]) -> int:
        n = len(nums)
        total = 0
        for i in range(1 << n):
            ss_xor = 0
            for j in range(n):
                if i & (1 << j):
                    ss_xor ^= nums[j]
            
            total += ss_xor
        
        return total

        

File: 1993-sum-of-all-subset-xor-totals/sum-of-all-subset-xor-totals.cpp
class Solution {
public:
    int subsetXORSum(vector<int>& nums) {
        int n = nums.size();

        int ans = 0;
        for (int i = 0; i < (1 << n); i++) {
            int xor_of_subset= 0;
            for(int j = 0; j < n; j++) 
                if ((1 << j) & i)
                    xor_of_subset ^= nums[j];
            ans += xor_of_subset;
        }

        return ans;


    }
};